MonoMod/MonoModder.cs

```diff
@@ -1212,6 +1216,110 @@ public virtual TypeReference FindType(string name, bool runtimeName)
+                (targetEvent?.RemoveMethod ?? srcEvent.RemoveMethod)?.SetPublic(true);
+                foreach (MethodDefinition method in targetEvent?.OtherMethods ?? srcEvent?.OtherMethods)
+                    method.SetPublic(true);
+            }
+
+            if (srcEvent.HasMMAttribute("Ignore"))
+            {
+                if (backing != null)
+                    backing.DeclaringType.Fields.Remove(backing); // Otherwise the backing field gets added anyway
+                if (srcEvent.AddMethod != null)
+                    propMethods?.Add(srcEvent.AddMethod);
+                if (srcEvent.RemoveMethod != null)
+                    propMethods?.Add(srcEvent.RemoveMethod);
+                foreach (MethodDefinition method in srcEvent.OtherMethods)
+                    propMethods?.Add(method);
+                return;
+            }
+
+            if (srcEvent.HasMMAttribute("Remove") || srcEvent.HasMMAttribute("Replace"))
+            {
+                if (targetEvent != null)
+                {
+                    targetType.Events.Remove(targetEvent);
+                    if (targetBacking != null)
+                        targetType.Fields.Remove(targetBacking);
+                    if (targetEvent.AddMethod != null)
+                        targetType.Methods.Remove(targetEvent.AddMethod);
+                    if (targetEvent.RemoveMethod != null)
+                        targetType.Methods.Remove(targetEvent.RemoveMethod);
+                    if (targetEvent.OtherMethods != null)
+                        foreach (MethodDefinition method in targetEvent.OtherMethods)
+                            targetType.Methods.Remove(method);
+                }
+                if (srcEvent.HasMMAttribute("Remove"))
+                    return;
+            }
+
+            if (targetEvent == null)
+            {
+                // Add missing event
+                EventDefinition newEvent = targetEvent = new EventDefinition(srcEvent.Name, srcEvent.Attributes, srcEvent.EventType);
+                newEvent.AddAttribute(GetMonoModAddedCtor());
+
+                newEvent.DeclaringType = targetType;
+                targetType.Events.Add(newEvent);
+
+                if (backing != null)
+                {
+                    FieldDefinition newBacking = new FieldDefinition(backingName, backing.Attributes, backing.FieldType);
+                    targetType.Fields.Add(newBacking);
+                }
+            }
+
+            foreach (CustomAttribute attrib in srcEvent.CustomAttributes)
+                targetEvent.CustomAttributes.Add(attrib.Clone());
+
+            MethodDefinition adder = srcEvent.AddMethod;
+            if (adder != null &&
+                (addMethod = PatchMethod(targetType, adder)) != null)
+            {
+                targetEvent.AddMethod = addMethod;
+                foreach (CustomAttribute attrib in srcEvent.CustomAttributes)
+                    addMethod.CustomAttributes.Add(attrib.Clone());
```

Comment:
@iamwyza I added those two lines for the get, set and other methods in the properties patcher. Someone requested for MonoMod to process attributes added to the property on its "children," too, but I can think of another way to solve that.

Feel free to remove those here and further down :)

Reply:
Ok.  I'll do that when i get off work, and re-test.  If it works I'll push those changes.  It makes sense as to why it was doing that now that I realize what the reasoning was for it.  It may indeed need to be handled differently, but for now, just cloning the attributes at the top level of the event rather than the backing methods should be sufficient.